### ISSUE TITLE
Support completion for table key specifier/constraint

### ISSUE THREAD

**Issue by mohanvive** [ballerina-platform/ballerina-lang#32886]
**Description:**
Key specifier and key constraints are two possible options which help to define the key/key type for the table. As of now, LS supports completions for key specifier at table type. However below completion suggestions are missing now. 

1) key constraint type completion suggestion when defining table type
2) key specifier field suggestion when defining the table constructor

**Comment by malinthar** [comment]
Hi @mohanvive,  Had a look of the mentioned cases. Key specifier field suggestion has to be improved. However, Key type constraint seems to be suggested on beta4-snapshot. 
![Screenshot from 2021-09-24 17-19-03](https://user-images.githubusercontent.com/35211477/134670000-46f98fc3-a96a-4cc9-b609-11a885ed2e6e.png)

Can you verify whether the above is still there?

**Comment by mohanvive** [comment]
Yes, it does suggest types for the table key constraint. However, it is not accurate because table key constraint type only can be a type of the fields which are defined as readonly. 

For example, if `firstName` is the only readonly field of `Employee`, then only possible valid suggestion type is `string`.

**Comment by malinthar** [comment]
> Yes, it does suggest types for the table key constraint. However, it is not accurate because table key constraint type only can be a type of the fields which are defined as readonly.
> 
> For example, if `firstName` is the only readonly field of `Employee`, then only possible valid suggestion type is `string`.
> @malinthar

I see. Thanks @mohanvive

**Comment by malinthar** [comment]
> Yes, it does suggest types for the table key constraint. However, it is not accurate because table key constraint type only can be a type of the fields which are defined as readonly.
> 
> For example, if `firstName` is the only readonly field of `Employee`, then only possible valid suggestion type is `string`.
> @malinthar

suppose a user needs to use a particular type that he has not defined in the row type parameter. In that case we should suggest typedefs and modules.To accommodate both the scenarios we can sort the completions so that the already defined readonly params are on top.  We do not remove completion items. Instead, we sort them. This is the practice that we follow in other contexts. WDYT @mohanvive ?

**Comment by mohanvive** [comment]
> > Yes, it does suggest types for the table key constraint. However, it is not accurate because table key constraint type only can be a type of the fields which are defined as readonly.
> > For example, if `firstName` is the only readonly field of `Employee`, then only possible valid suggestion type is `string`.
> > @malinthar
> 
> suppose a user needs to use a particular type that he has not defined in the row type parameter. In that case we should suggest typedefs and modules.To accommodate both the scenarios we can sort the completions so that the already defined readonly params are on top. This is the practice that we follow in other contexts. WDYT @mohanvive ?

Yes, that makes sense. +1.
